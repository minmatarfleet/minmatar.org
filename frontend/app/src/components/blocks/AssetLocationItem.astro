---
import { i18n } from '@helpers/i18n'
const { t } = i18n(Astro.url)

import type { AssetLocationItemUI } from '@dtypes/layout_components'

interface Props {
    asset_location:  AssetLocationItemUI;
}

const {
    asset_location,
} = Astro.props

import Flexblock from '@components/compositions/Flexblock.astro';
---

<Flexblock
    gap='0'
    class="[ asset-location-item ] [ w-full ]"
>
    <span class="text-[var(--highlight)]">
        {asset_location.location_name}
    </span>
    <small>
        {asset_location.assets_count} {asset_location.assets_count != 1 ? t('assets').toLowerCase() : t('asset').toLowerCase()} {t('in_inventory')}
    </small>
</Flexblock>