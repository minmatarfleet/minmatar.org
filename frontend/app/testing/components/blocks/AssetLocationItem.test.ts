import { experimental_AstroContainer as AstroContainer } from "astro/container";
import { expect, test } from "vitest";
import AssetLocationItem from "@components/blocks/AssetLocationItem.astro";

test("AssetLocationItem defaults", async () => {
  const container = await AstroContainer.create();
  const result = await container.renderToString(AssetLocationItem, {
    props: {
      asset_location: {
        location_name: "chick fil a",
        assets_count: 9871,
      },
    },
  });

  expect(result).toMatchSnapshot();
});
