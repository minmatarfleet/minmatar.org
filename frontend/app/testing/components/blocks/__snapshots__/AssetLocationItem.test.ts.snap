// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AssetLocationItem defaults 1`] = `"<div class="flex-block [ asset-location-item ] [ w-full ] gap-0" data-mod style="--flex_block_gap: 0;">  <span class="text-[var(--highlight)]"> chick fil a </span> <small> 9871 assets in inventory </small>  </div> "`;
